src/Voron/Data/CompactTrees/CompactTree.cs

```diff
@@ -1080,18 +1092,16 @@ private void InsertToStack(CursorState newPageState)
             _internalCursor._pos++;
         }
 
-        private void DefragPage()
+        private static void DefragPage(LowLevelTransaction llt, ref CursorState state)
         {                     
-            ref var state = ref _internalCursor._stk[_internalCursor._pos];
-
-            using (_llt.Environment.GetTemporaryPage(_llt, out var tmp))
+            using (llt.Environment.GetTemporaryPage(llt, out var tmp))
             {
-                var tmpHeader = (CompactPageHeader*)tmp.TempPagePointer;
+                // Ensure we clean up the page.               
+                Unsafe.InitBlock(tmp.TempPagePointer, 0, Constants.Storage.PageSize);
```

Comment:
That is faster?

Reply:
Same as `.CopyBlockUnaligned()` and `.CopyBlock()`